xarray/tests/test_merge.py

```diff
@@ -358,7 +371,7 @@ def test_merge_broadcast_equals(self):
 
         ds1 = xr.Dataset({"x": np.nan})
         ds2 = xr.Dataset({"x": ("y", [np.nan, np.nan])})
-        actual = ds1.merge(ds2)
+        actual = ds1.merge(ds2, compat="no_conflicts")
```

Comment:
I think everything in this test should be `compat="broadcast_equals"` I believe that is the intent.